vault/token_store.go

```diff
@@ -746,6 +760,64 @@ func (ts *TokenStore) tokenStoreAccessorList(ctx context.Context, req *logical.R
 	return resp, nil
 }
 
+func (ts *TokenStore) tokenStoreAccessorListDetails(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	ns, err := namespace.FromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	nsID := ns.ID
+
+	entries, err := ts.accessorView(ns).List(ctx, "")
+	if err != nil {
+		return nil, err
+	}
+
+	resp := &logical.Response{}
+	var ret []map[string]interface{}
+
+	for _, entry := range entries {
+		aEntry, err := ts.lookupByAccessor(ctx, entry, true, false)
+		if err != nil {
+			resp.AddWarning(fmt.Sprintf("Found an accessor entry that could not be successfully decoded; associated error is %q", err.Error()))
+			continue
+		}
+
+		if aEntry.TokenID == "" {
+			resp.AddWarning(fmt.Sprintf("Found an accessor entry missing a token: %v", aEntry.AccessorID))
+			continue
+		}
+
+		if aEntry.NamespaceID == nsID {
+			info := map[string]interface{}{
+				"accessor_id":        aEntry.AccessorID,
+				"token_display_name": aEntry.TokenDisplayName,
+				"token_role":         aEntry.TokenRole,
+				"token_ttl":          int64(0),
+				"token_policies":     aEntry.TokenPolicies,
+			}
+
+			tokenEntry, _ := ts.lookupInternal(ctx, aEntry.TokenID, false, true)
```

Comment:
Since we are doing the tokenEntry lookup here to check the lease times, could we maybe use the displayName/Policies/Role info from that object to populate the response, instead of fetching them from the accessorEntry object ?

Reply:
That's a great point, I thought of going with this approach originally. At first I was wondering if there was a way to check the lease times without the tokenEntry (e.g. generate a lease ID and store it in the accessorEntry object), but after going through the logic in `expiration.go`, I don't think it is possible. So I will use the info from tokenEntry to populate the response, thanks for catching this! 👍🏽

Reply:
no worries! yeah I was thinking it might be possible to then get away with not having any changes in the accessorEntry struct, as the info kinda duplicates the information in tokenEntry so was wondering if it might be possible to avoid that, I think an extra read from storage would be a totally fine tradeoff to make.